### PR TITLE
[JBPM-9677] Invalid session was used for this context when aborting process instance with active exception handling sub process

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9677

### CODE REVIEW

**Review by afalhambra — APPROVED** [review]
LGTM - just some cosmetic changes

File: jbpm-test-coverage/src/test/java/org/jbpm/test/functional/workitem/WorkItemExceptionHandlingStrategyTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        );
+        RuntimeEngine runtimeEngine = getRuntimeEngine( ProcessInstanceIdContext.get() );
+        KieSession kieSession = runtimeEngine.getKieSession();
+
+        kieSession.getWorkItemManager().registerWorkItemHandler( "Human Task", new ExceptionWorkItemHandler(EXCEPTION_HANDLING_PROCESS_ID, HandlingStrategy.ABORT.name()) );
+        ProcessInstance pi = kieSession.startProcess( FAILING_PROCESS_ID );
+        assertProcessInstanceActive( pi.getId(), kieSession );
+
+        kieSession.abortProcessInstance(pi.getId());
+        assertProcessInstanceAborted( pi.getId(), kieSession );
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-test-coverage/src/test/java/org/jbpm/test/functional/workitem/WorkItemExceptionHandlingStrategyTest.java:49]: changed to assertProcessInstanceAborted

File: jbpm-test-coverage/src/test/java/org/jbpm/test/functional/workitem/WorkItemExceptionHandlingStrategyTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    private static final String FAILING_PROCESS_ID = "com.sample.boolean.waitSignal";
+    private static final String EXCEPTION_HANDLING_PROCESS_ID = "WaitSignal";
+
+    public WorkItemExceptionHandlingStrategyTest() {
+        super( true, true );
+    }
+
+    @Test
+    public void exceptionHandlingAbortProcess() {
+        manager = createRuntimeManager(Strategy.PROCESS_INSTANCE, (String) null,
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-test-coverage/src/test/java/org/jbpm/test/functional/workitem/WorkItemExceptionHandlingStrategyTest.java:37]: not needed

File: jbpm-test-coverage/src/test/java/org/jbpm/test/functional/workitem/ExceptionWorkItemHandler.java [hunk 3]
```diff
@@ -18,23 +18,38 @@
...
-        try {
-            if ( "no".equals( exception ) ) {
-                manager.completeWorkItem( workItem.getId(), null );
-            }
-            else {
-                throwExceptionSoThatWorkItemIsNOTCompleted( workItem );
+        Object exception = workItem.getParameter( "exception" );
+        if (handlingProcessId == null && handlingStrategy == null) {
+            try {
+                if ( exception != null && "no".equals( (String)exception ) ) {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-test-coverage/src/test/java/org/jbpm/test/functional/workitem/ExceptionWorkItemHandler.java:41]: not needed.

**Pull request merged**